ui-admin/src/user/DeleteUserModal.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react'
+import { Store } from 'react-notifications-component'
+import Api, { AdminUser, Portal } from 'api/api'
+import { failureNotification, successNotification } from 'util/notifications'
+
+/** Modal to remove a user from a portal. */
+const DeleteUserModal = ({ subjUser, portal, userDeleted, onDismiss }:
+                           {subjUser: AdminUser, portal: Portal, userDeleted: () => void, onDismiss: () => void}) => {
+  const [usernameConfirm, setUsernameConfirm] = useState('')
+  const removeString = `remove ${subjUser.username}`
+  const canRemove = usernameConfirm === removeString
+
+  const doRemove = async () => {
+    try {
+      await Api.removePortalUser(subjUser, portal.shortcode)
+      Store.addNotification(successNotification(`${subjUser.username} removed`))
+      userDeleted()
+      onDismiss()
+    } catch (e) {
+      console.log('error removing user', e)
+      Store.addNotification(failureNotification('Error removing user'))
+    }
+  }
+
+  return <div className="p4">
+    <form onSubmit={e => e.preventDefault()}>
+      <h3 className="h5">Remove user from portal: {subjUser.username}</h3>
+      <div className="my-3">
+        <label>
+          Confirm by typing &quot;{removeString}&quot; below<br/>
+          <strong>Removing a user is permanent</strong>
```

Comment:
I don't know that we need to be so strong about this disclaimer -- since they can add them back relatively easily.  This might make people think they can't ever re-add them--right now there's no data that would be lost.

Reply:
Will change.
Was following the lead of similar dialog, but had the same concerns.